### DISCUSSION TITLE
would love some help with io_uring_register_buffers_tags

### DISCUSSION THREAD

**Discussion by romange** [axboe/liburing#1038] [Q&A] [answered]
I am trying to understand how I can leverage the enhanced capabilities of tags API. 
What is the suggested flow when registering/updating tagged resources?  
How do you use tags in your applications?

**Comment by isilence** [comment, marked as answer]
When you unregister a buffer (e.g. `IORING_REGISTER_BUFFERS_UPDATE`), it's not released immediately and might still be in use after the update reported success (syscall returns), and so you wouldn't know when it's safe to reuse the buffer. With tags io_uring will post a CQE when it has actually been released.
Tags are mainly there for registered buffers. They work for registered files, but they don't have such a problem.

**Reply by romange** [reply]
Thanks, I completely misunderstood the intention of tags. Now it's much clearer.
